### PR TITLE
[CI]updated drone.star file to match the latest changes

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This is an update on drones.star file 
- using drone-cancel-previous-builds
- using thegeeklab/drone-github-comment instead of jmccann/drone-github-comment for github comments 
- moving docker image references to constants
- using caches images
- Do not run PHP unit tests with Oracle by default

## Related Issue
- Part of issue: 
- https://github.com/owncloud/QA/issues/700
- https://github.com/owncloud/QA/issues/717
- https://github.com/owncloud/QA/issues/729
- https://github.com/owncloud/QA/issues/728
- https://github.com/owncloud/core/issues/38679

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
lgtm

**Comment by phil-davis** [comment]
The `skipIfUnchanged` changes are also needed. It should be easy to copy the starlark code from the current activity master branch, and get it all up-to-date in a single PR. (And do the same for all the other oC10 app repos)

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/calendar/1027
`.drone.star:11:1: cannot reassign global OC_CI_NODEJS declared at .drone.star:10:1`

File: .drone.star [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+MAILHOG = "mailhog/mailhog"
+MINIO_MC_RELEASE_2020_VERSION = "minio/mc:RELEASE.2020-12-18T10-53-53Z"
 OC_CI_ALPINE = "owncloudci/alpine:latest"
-OC_CI_WAIT_FOR = "owncloudci/wait-for:latest"
+OC_CI_BAZEL_BUILDIFIER = "owncloudci/bazel-buildifier"
+OC_CI_CEPH_UBUNTU = "owncloudci/ceph:tag-build-master-jewel-ubuntu-16.04"
+OC_CI_CORE = "owncloudci/core"
+OC_CI_DRONE_CANCEL_PREVIOUS_BUILDS = "owncloudci/drone-cancel-previous-builds"
+OC_CI_NODEJS = "owncloudci/nodejs:%s"
 OC_CI_NODEJS = "owncloudci/nodejs:14"
```
Comment (amrita-shrestha) [inline .drone.star]: remove this line
```suggestion

```

**Comment by SagarGi** [comment]
@sushmita56  please wait for activity app for a refactor.

File: .drone.star [hunk 2]
```diff
@@ -151,16 +175,16 @@ def codestyle(ctx):
...
+                "steps": skipIfUnchanged(ctx, "lint") +
+                         [
+                             {
+                                 "name": "coding-standard",
+                                 "image": OC_CI_PHP % phpVersion,
+                                 "commands": [
+                                     "make test-php-style",
+                                 ],
+                             },
+                         ],
```
Comment (SwikritiT) [inline .drone.star:188]: The indentation looks a little bit off here please fix it

Comment (SwikritiT) [inline .drone.star:188]: The indentation looks a little bit off here please fix it

File: .drone.star [hunk 3]
```diff
@@ -195,16 +219,16 @@ def jscodestyle(ctx):
...
+        "steps": skipIfUnchanged(ctx, "lint") +
+                 [
+                     {
+                         "name": "coding-standard-js",
+                         "image": OC_CI_NODEJS,
+                         "commands": [
+                             "make test-js-style",
+                         ],
+                     },
+                 ],
```
Comment (SwikritiT) [inline .drone.star]: Here too

Comment (SwikritiT) [inline .drone.star]: Here too

**Comment by phil-davis** [comment]
When I diff with the code in activity app I get:
```
$ git diff
diff --git a/.drone.star b/.drone.star
index 8d61e968..c1e04ceb 100644
--- a/.drone.star
+++ b/.drone.star
@@ -25,8 +25,8 @@ SELENIUM_STANDALONE_FIREFOX_DEBUG = "selenium/standalone-firefox-debug:3.8.1"
 SONARSOURCE_SONAR_SCANNER_CLI = "sonarsource/sonar-scanner-cli"
 THEGEEKLAB_DRONE_GITHUB_COMMENT = "thegeeklab/drone-github-comment:1"
 
-DEFAULT_NODEJS_VERSION = "14"
 DEFAULT_PHP_VERSION = "7.4"
+DEFAULT_NODEJS_VERSION = "14"
 
 dir = {
     "base": "/var/www/owncloud",
@@ -1542,7 +1542,7 @@ def getDbDatabase(db):
 def getNodeJsVersion():
     if "nodeJsVersion" not in config:
         # We use nodejs 14 as the default
-        return "14"
+        return DEFAULT_NODEJS_VERSION
     else:
         return config["nodeJsVersion"]
 
@@ -1968,7 +1968,7 @@ def githubComment(earlyFail):
             "pull": "if-not-exists",
             "settings": {
                 "message": ":boom: Acceptance tests pipeline <strong>${DRONE_STAGE_NAME}</strong> failed. The build has been cancelled.\\n\\n${DRONE_BUILD_LINK}/${DRONE_JOB_NUMBER}${DRONE_STAGE_NUMBER}",
-                "key": "pr-${DRONE_PULL_REQUEST}",  #TODO: we could delete the comment after a successful CI run
+                "key": "pr-${DRONE_PULL_REQUEST}",
                 "update": "true",
                 "api_key": {
                     "from_secret": "github_token",
```

Please adjust here so that the code is exactly the same as in the activity app. @sushmita56

**Comment by grgprarup** [comment]
@sushmita56 `.drone/cancelBuilds.sh` can be removed after `.drone.star` is updated.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**